pkg/handler/collector/github/github.go

```diff
@@ -0,0 +1,158 @@
+package github
+
+import (
+	"context"
+	"errors"
+	"io"
+	"net/http"
+	"net/url"
+	"strings"
+	"time"
+
+	"github.com/google/go-github/github"
+	"github.com/guacsec/guac/pkg/handler/processor"
+	"github.com/guacsec/guac/pkg/logging"
+	"go.uber.org/zap"
+	"golang.org/x/oauth2"
+)
+
+const (
+	CollectorGitHubDocument = "GitHubCollector"
+)
+
+type githubDocumentCollector struct {
+	poll     bool
+	interval time.Duration
+	token    string
+	owner    string
+	repo     string
+	tag      string
+	tagList  []string
+}
+
+func NewGitHubDocumentCollector(ctx context.Context, poll bool, interval time.Duration, logger *zap.SugaredLogger, token string, owner string, repo string, tag string, tagList []string) *githubDocumentCollector {
+	return &githubDocumentCollector{
+		poll:     poll,
+		interval: interval,
+		token:    token,
```

Comment:
Should we just get the token from an environment variable rather than passing it in?

Reply:
@pxp928 something like what we did with `os.Getenv("API_KEY")`?

Reply:
I don't think we should. I think the caller of the githubDocumentCollector should pass it in, otherwise the functions in the githubDocumentCollector need to collect info from the OS environment instead of having the caller do it and we can encapsulate that well somewhere else.